teaser/src/main/kotlin/com/mapbox/vision/teaser/recorder/RecordingActivity.kt

```diff
@@ -0,0 +1,121 @@
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import kotlinx.android.synthetic.main.activity_recording.*
+
+class RecordingActivity : BaseVisionActivity() {
+
+    companion object {
+        private val dateFormat = SimpleDateFormat("yyyy-MM-dd_HH-mm-ssZ", Locale.US)
+    }
+
+    private var isPermissionsGranted = false
+    private var visionManagerWasInit = false
+
+    private var modelPerformance = ModelPerformance.Off
+
+    private val visionEventsListener = object : VisionEventsListener {
+
+        override fun onCountryUpdated(country: Country) {
+        }
+
+        override fun onCameraUpdated(camera: Camera) {
+            runOnUiThread {
+                fps_performance_view.setCalibrationProgress(camera.calibrationProgress)
+            }
+        }
+
+        @SuppressLint("SetTextI18n")
+        override fun onUpdateCompleted() {
+            val frameStatistics = VisionManager.getFrameStatistics()
+            runOnUiThread {
+                fps_performance_view.showInfo(frameStatistics)
+            }
+        }
+    }
+
+    override fun setLayout() {
+        setContentView(R.layout.activity_recording)
+    }
+
+    override fun onPermissionsGranted() {
+        isPermissionsGranted = true
+
+        root.setOnLongClickListener {
+            if (fps_performance_view.visibility == View.GONE) {
+                fps_performance_view.show()
+            } else {
+                fps_performance_view.hide()
+            }
+            return@setOnLongClickListener true
+        }
+        recording_view.setOnClickListener { toggleRecording() }
+        fps_performance_view.hide()
+
+        tryToInitVisionManager()
+    }
+
+    private fun tryToInitVisionManager() {
+        if (isPermissionsGranted && !visionManagerWasInit) {
+            VisionManager.create()
+            VisionManager.start()
+            VisionManager.visionEventsListener = visionEventsListener
+            VisionManager.setModelPerformance(modelPerformance)
+            vision_view.setVisionManager(VisionManager)
+
+            visionManagerWasInit = true
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        tryToInitVisionManager()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        vision_view.onResume()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        vision_view.onPause()
+    }
+
+    override fun onStop() {
+        super.onStop()
+
+        if (isPermissionsGranted && visionManagerWasInit) {
+            VisionManager.stopRecording()
+            VisionManager.stop()
+            VisionManager.destroy()
+            visionManagerWasInit = false
+        }
+    }
+
+    private fun toggleRecording() {
+        when (recording_view.state) {
+            RecordingView.State.NotRecording -> {
+                VisionManager.startRecording("$BASE_SESSION_PATH/${dateFormat.format(Date(System.currentTimeMillis()))}")
```

Comment:
how about small function for `${dateFormat.format(Date(System.currentTimeMillis()))}` ?

Reply:
fixed